server/notification-providers/sevenio.js

```diff
@@ -0,0 +1,77 @@
+const NotificationProvider = require("./notification-provider");
+const Axios = require("axios");
+const { DOWN, UP } = require("../../src/util");
+
+class SevenIO extends NotificationProvider {
+
+    name = "SevenIO";
+
+    /**
+     * @inheritdoc
+     */
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        const okMsg = "Sent Successfully.";
+
+        const data = {
+            to: notification.sevenioTo,
+            from: notification.sevenioSender || "Uptime Kuma",
+            text: msg,
+        };
+
+        const axios = Axios.create({
+            baseURL: "https://gateway.seven.io/api/",
+            headers: {
+                "Content-Type": "application/json",
+                "X-API-Key": notification.sevenioApiKey,
+            },
+        });
+
+        try {
+            // testing or certificate expiry notification
+            if (heartbeatJSON == null) {
+                await axios.post("sms", data);
+                return okMsg;
+            }
+
+            let address;
+
+            switch (monitorJSON["type"]) {
+                case "ping":
+                    address = monitorJSON["hostname"];
+                    break;
+                case "port":
+                case "dns":
+                case "gamedig":
+                case "steam":
+                    address = monitorJSON["hostname"];
+                    if (monitorJSON["port"]) {
+                        address += ":" + monitorJSON["port"];
+                    }
+                    break;
+                default:
+                    if (![ "https://", "http://" ].includes(monitorJSON["url"])) {
+                        address = monitorJSON["url"];
+                    } else {
+                        address = "n/a";
```

Comment:
nit:
I think adding ` (n/a)` to the sms should not be done in this case as this will likely just confuse users.
I think extracting this into a function and don't add the address in this case would be better.